service.py

```diff
@@ -140,28 +176,37 @@ def createListItemFromFlatPlaylistItem(video):
 ydl.add_default_info_extractors()
 
 with ydl:
-    showInfoNotification("resolving stream(s) for " + url)
+    showInfoNotification("Resolving stream(s) for " + url)
     result = ydl.extract_info(url, download=False)
 
 if 'entries' in result:
     # more than one video
     pl = xbmc.PlayList(1)
     pl.clear()
 
-    # make sure first ListItem has a resolved url, to avoid recursion and Kodi crashes
-    firstVideoUrl = result['entries'][0]['url']
-    firstItem = createListItemFromVideo(ydl.extract_info(firstVideoUrl, download=False))
-    pl.add(firstItem.getPath(), firstItem)
-    
-    for video in result['entries'][1:]:
+    # jump in the queue to the video specified in the url
+    indexToStartAt = playlistIndex(url, result)
+    if indexToStartAt == None:
+        indexToStartAt = 0
+
+    unresolvedEntries = result['entries']
+    startingEntry = unresolvedEntries.pop(indexToStartAt)
+
+    # populate the queue with unresolved entries so that the starting entry can be inserted
+    for video in unresolvedEntries:
         list_item = createListItemFromFlatPlaylistItem(video)
         pl.add(list_item.getPath(), list_item)
 
+    # make sure the starting ListItem has a resolved url, to avoid recursion and crashes
+    startingVideoUrl = startingEntry['url']
+    startingItem = createListItemFromVideo(ydl.extract_info(startingVideoUrl, download=False))
+    pl.add(startingItem.getPath(), startingItem, indexToStartAt)
+    
     #xbmc.Player().play(pl) # this probably works again
     # ...but start playback the same way the Youtube plugin does it:
-    xbmc.executebuiltin('Playlist.PlayOffset(%s,%d)' % ('video', 0))
+    xbmc.executebuiltin('Playlist.PlayOffset(%s,%d)' % ('video', indexToStartAt))
 
-    showInfoNotification("playing playlist " + result['title'])
+    showInfoNotification("Playing playlist " + result['title'])
```

Comment:
Btw. is there always a `title` key for the playlist? Might we run in the same issue here that we had with missing video titles in playlist resolving on demand PR? https://github.com/firsttris/plugin.video.sendtokodi/blob/e1fa30fc303c95b72f6765aca6e70314be5e5777/service.py#L90

Reply:
I haven't experienced that myself, but I think that even the generic resolver will always be able to fall back on the page's title. 

Don't know what happens when a page is missing a title though. Empty string? URL? But I think we can leave that corner case for now.